modules/core/src/lib/layer.js

```diff
@@ -679,23 +699,21 @@ export default class Layer extends Component {
 
   // Calculates uniforms
   drawLayer({moduleParameters = null, uniforms = {}, parameters = {}}) {
+    const currentProps = this.props;
     if (!uniforms.picking_uActive) {
-      this.updateTransition();
+      // Overwrite this.props during redraw to use in-transition prop values
+      this.props = this.updateTransition();
     }
 
+    const {opacity} = this.props;
+    // apply gamma to opacity to make it visually "linear"
+    uniforms.opacity = Math.pow(opacity, 1 / 2.2);
```

Comment:
Opacity can potentially be in transition.

Reply:
I understand that, but why is it gamma corrected here? I don't remember seeing gamma correction anywhere else in deck.

Reply:
This is existing logic copied from the current `_updateBaseUniforms` method. @ibgreen

Reply:
Yes, we do apply this gamma to opacity before passing it to the shader. 

This is just the classic gamma correction to make the opacity perceptually linear on typical non-linear computer screens.

Try playing with an opacity slider in a layer (perhaps in the layer browser) with and without this correction...

Reply:
We can discuss in detail elsewhere, since it's unrelated to this PR, but:
- gamma correction is a color correction. It isn't meant for opacity (see [e.g.](https://learnopengl.com/Advanced-Lighting/Gamma-Correction)).
- If you aren't transitioning to 0 or 1, you'll end up on the wrong value.

Reply:
> gamma correction is a color correction. It isn't meant for opacity

I think I get what you are saying. 

To be clear, this was not added because of an incorrect assumption that this is the "blessed or correct way" to handle opacity. 

It was added after experimentation because it gives noticeable improvements in the behavior of this prop. 

We essentially use it as a brightness correction heuristic (which actually seems consistent with the linked article). 

Have nothing against replacing it with some other adjustment, but I would not be in favor of simply removing the correction as it solves a problem.

> If you aren't transitioning to 0 or 1, you'll end up on the wrong value.

If that is the case it would be a bug in this PR. The gamma adjustment should not affect the prop value being transitioned, only the derived uniform values being passed to the shader.